### DISCUSSION TITLE
Can build calculation

### DISCUSSION THREAD

**Discussion by HannesMorre** [inventree/InvenTree#5517] [Q&A]
Hi All

As you can see in the picture below we have several assemblies that are made of subassemblies which in turn are made of more subassemblies and so on. In the top right you can see that we can build 127 pieces of this part, since we have 127 pieces of the first bom item and 911 of the second item. But in fact we can build 668 +127 pieces of this part, because we have 668 parts needed for the subassembly of which we have 127 in stock now. Is it possible to show the actual "can build" amount for this part or not.
![image](https://github.com/inventree/InvenTree/assets/71696745/f6f8bd69-79c9-4722-8f4d-aa4eba4310a7)

I know my explanation is a bit clunky but I hope you guys understand. What I want to achieve is that all parts of child boms are considered to calculate the "can build" amount of the finished product.

Is this in any way possible?

Thanks!

**Comment by SchrodingersGat** [comment]
Possible, yes, but not easy (or simple).

The "can build" calculation is relatively simple in terms of database access. However if you start drilling down into sub-assemblies and sub-sub-assemblies this very quickly becomes expensive and complicated to compute.

Further, the further you go down, the more complex it becomes in terms of stock allocation - you might calculate that you can build 100, but a sub-sub-sub part is actually allocated to make something else. 

To calculate this "in real time" would be too prohibitive. However it could be achievable to perform this task "on demand" for a specific part. Perhaps a "build availability report" which could be requested for a single part, and then the necessary calculations performed, and a report (maybe a .csv?) could be delivered to the user.

**Comment by HannesMorre** [comment]
I agree that these calculations would be too complex to calculate in real time, but a build availability report would be nice to have. The reason I'm asking is because we built a forecasting application based on offers, sales orders and stock information from Inventree. It would be useful to add build availability data to this including all sub items.
